From m.kozlowski@tuxland.pl  Wed Mar 11 14:35:16 2009
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Date: Tue,  3 Mar 2009 19:38:34 +0100
Subject: Staging: comedi: usbdux: remove .bss variable initialization
To: gregkh@suse.de
Cc: m.kozlowski@tuxland.pl
Message-ID: <1236105518-19565-5-git-send-email-m.kozlowski@tuxland.pl>


This patch removes explicit zeroing of usbduxsub
variable on init because it is in .bss section.

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Cc: Bernd Porr <BerndPorr@f2s.com>
Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/usbdux.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/usbdux.c
+++ b/drivers/staging/comedi/drivers/usbdux.c
@@ -2917,10 +2917,8 @@ static void init_usb_devices(void)
 	/* all devices entries are invalid to begin with */
 	/* they will become valid by the probe function */
 	/* and then finally by the attach-function */
-	for (index = 0; index < NUMUSBDUX; index++) {
-		memset(&(usbduxsub[index]), 0x00, sizeof(usbduxsub[index]));
+	for (index = 0; index < NUMUSBDUX; index++)
 		init_MUTEX(&(usbduxsub[index].sem));
-	}
 }
 
 /* Table with the USB-devices: just now only testing IDs */
